src/fft.jl

```diff
@@ -253,14 +255,16 @@ _fftw_flags(::Type{Float64}) = FFTW.MEASURE
 Plan a FFT of type `T` and size `fft_size`, spending some time on finding an
 optimal algorithm. (Inplace, out-of-place) x (forward, backward) FFT plans are returned.
 """
-function build_fft_plans(T::Union{Type{Float32}, Type{Float64}}, fft_size)
-    tmp = Array{Complex{T}}(undef, fft_size...)
-    ipFFT = FFTW.plan_fft!(tmp, flags=_fftw_flags(T))
-    opFFT = FFTW.plan_fft(tmp, flags=_fftw_flags(T))
+#Removed the flags as CUDA's plan_fft doesn't need flags. If this is a performance issue, we should check array_type's type then call either FFTW.plan_fft(tmp, flags = ...) or CUDA.plan_fft(tmp)
+function build_fft_plans(array_type::AbstractArray{T}, fft_size) where {T<:Union{Float32,Float64}}
+    tmp = similar(array_type, Complex{Float64}, fft_size...)
+    ipFFT = AbstractFFTs.plan_fft!(tmp)
+    opFFT = AbstractFFTs.plan_fft(tmp)
     # backward by inverting and stripping off normalizations
     ipFFT, opFFT, inv(ipFFT).p, inv(opFFT).p
 end
 
+LinearAlgebra.mul!(Y::AbstractGPUArray, p::AbstractFFTs.Plan, X::AbstractGPUArray) = Y .= p *X
```

Comment:
Ooof that's very bad for performance. Is this really needed?

Reply:
I don't think it's bad for performance, as it is actually what is recommended by the [AbstractFFT documentation](https://juliamath.github.io/AbstractFFTs.jl/stable/api/#AbstractFFTs.plan_fft). We should either use `*` or use the `mul! `function: the thing is there are a few restrictions to use `mul!` (the input array must be a complex floating-point array like the output). 
Initially I hadn't understood that, so that's why there is this weird workaround. But now I fixed the code, this is no longer useful, and I am removing it. Thanks for pointing that out!

Reply:
Great! Basically any time there's a `!` it should mean it's a place where performance and allocations matter (otherwise, people should just use the non-mutating version, as it's clearer, easier to reason with and hides fewer bugs), so what you do here (which allocates a new array for p*X) breaks that implicit contract